packages/augur-core/source/contracts/reporting/RepPriceOracle.sol

```diff
@@ -1,25 +1,35 @@
-pragma solidity 0.5.10;
+pragma solidity 0.5.15;
 
 import 'ROOT/IAugur.sol';
 import 'ROOT/libraries/Initializable.sol';
 import 'ROOT/uniswap/interfaces/IUniswapV2.sol';
 import 'ROOT/uniswap/interfaces/IUniswapV2Factory.sol';
 import 'ROOT/reporting/IRepPriceOracle.sol';
+import 'ROOT/ens/IENSRegistry.sol';
+import 'ROOT/ens/IENSResolver.sol';
 
 
 contract RepPriceOracle is IRepPriceOracle, Initializable {
 
+    uint256 constant Q112 = 2**112;
+
+    bytes32 public constant UNISWAP_REGISTRY_ENS_NAME = 0xf259e1e59b6e1e9de21a18289a53332ebc255d2e4ece8a91e0e90c628a9c7f87; // "uniswapv2.eth"; // TODO: set this to the actual namehash of the owned ENS name
+
     IAugur public augur;
     address public cash;
-    uint256 public tau = 1 days / 15; // The time period (in blocks) of the twa, assuming a block is mined every 15 seconds. TODO: Review this value
+    uint256 public period = 3 days; // TODO: revisit if this is an appropriate period
     IUniswapV2Factory public uniswapFactory;
+    IENSRegistry public ensRegistry;
+    bool public uniswapUpgraded;
 
     struct ExchangeData {
         IUniswapV2 exchange;
-        uint256 cashAmountAccumulated;
-        uint256 repAmountAccumulated;
+        uint256 repPriceAccumulated;
         uint256 blockNumber;
+        uint256 blockTimestamp;
```

Comment:
What is the purpose of storing the timestamp?  Just for convenience for integrating applications?  It doesn't appear to be read anywhere in this contract.

Reply:
its used to weigh the accumulated delta if the amount of time passed since last update is less than the period